src/devices/machine/ncr5385.cpp

```diff
@@ -761,6 +809,36 @@ int ncr5385_device::state_step()
 			// clear data and ACK
 			scsi_bus->data_w(scsi_refid, 0);
 			scsi_bus->ctrl_w(scsi_refid, 0, S_ACK);
+
+			if (m_cnt == 0)
+			{
+#ifdef USE_TEK4404_CHANGES
```

Comment:
I've tested simulating a slow hard disk instead of this change, and it can successfully complete the "diskrepair" and reach a login prompt. I don't have the knowledge of this system to test further, but I believe this change is wrong and the delay occur in the target, not this device.